proposals/materialx-versioning/README.md

```diff
@@ -0,0 +1,130 @@
+From its inception, the MaterialX project has considered MaterialX documents to be an archival format.  What goes along 
+with this is an implied promise that opening a MaterialX file from 10+ years ago should generate a material that will 
+render a qualitatively similar image.  This promise is honored by the project in the form of an automatic system that 
+upgrades data as documents are loaded to be compatible with the current MaterialX library.
+
+The MaterialX library is a collection of XML files that describe a standard set of node definitions, type definitions and 
+other components provided by MaterialX.  The Material library version is a `major.minor` version number (currently 1.38) 
+and allows that set of definitions to be versioned.  It is not the version of the MaterialX project itself, which also 
+includes a `patch` version number, but the MaterialX library version does track in-sync with the MaterialX project version.
+
+The MaterialX integration within OpenUSD relies upon two libraries, `UsdMtlx` and `HdMtlx`. `UsdMtlx` transcodes the 
+MaterialX document in to `UsdShade` primitives. Inside of Hydra, `HdMtlx` is used to reconstruct the
+MaterialX document using the `UsdShade` primitives that have been transported through Hydra (more information 
+available [here](https://openusd.org/release/api/_page__material_x__in__hydra__u_s_d.html)). The reconstructed MaterialX 
+document is then passed to the MaterialX Shader Generation library to generate shader code in the desired destination
+shading languages.
+
+## Problem
+For the entire lifetime of MaterialX support within OpenUSD, there has never been a MaterialX library version change. The 
+MaterialX library version has always been v1.38. The MaterialX project would like to be able to update the MaterialX 
+library version without invalidating MaterialX data currently stored in OpenUSD files. Without access to the MaterialX 
+library version the upgrade code path inside the MaterialX project cannot be used.
+
+Without some sort of remedy, either MaterialX will be constrained to stay at version 1.38, or OpenUSD will become a 
+container for ambiguous MaterialX data.  
+
+## Proposal to Version MaterialX data in OpenUSD
+### Overview
+We propose adding an applied API schema `MaterialXInfoAPI` (in the spirit of the 
+["Revise use of Layer Metadata in USD"](https://github.com/PixarAnimationStudios/OpenUSD-proposals/blob/main/proposals/revise_use_of_layer_metadata/README.md)) that can record the MaterialX library version. We keep the name of the 
+applied API schema a little more generalized to leave the door open for additional MaterialX document information to be
+added to the schema at a later date. We deliberately decide to only propose adding the MaterialX library version to the 
+schema to keep the proposal as uncontroversial as possible to add rapid adoption.
+
+The `MaterialXInfoAPI` will be applied to the UsdShade `Material` primitive. This allows for referencing the material 
```

Comment:
If the aim is to only restrict this APIs application to UsdShadeMaterial, then we should also consider "apiSchemaCanOnlyApplyTo = UsdShadeMaterial". Refer documentation for the same here: https://openusd.org/dev/api/_usd__page__generating_schemas.html

Reply:
Plus to this... will prevent ambiguous applications of putting versions directly on Shader prims or other prims, and restrictions can be lifted later, but not the other way round.  Question, though: will NodeGraphs need to be versioned independently within a document (e.g. if they are referenced from another mtlx file, or can we assume that every mtlx document referenced onto a UsdStage is internally version-consistent?